bson/src/main/org/bson/types/Decimal128.java

```diff
@@ -374,6 +373,84 @@ public boolean isNaN() {
         return (high & NaN_MASK) == NaN_MASK;
     }
 
+
```

Comment:
This is the main work here.  These comments were adapted from the corresponding ones in the BigDecimal class.

Reply:
Ack